gobblin-data-management/src/main/java/org/apache/gobblin/data/management/conversion/hive/source/HiveSource.java

```diff
@@ -276,6 +279,11 @@ protected void createWorkunitForNonPartitionedTable(HiveDataset hiveDataset, boo
     }
   }
 
+  @Deprecated
+  protected HiveWorkUnit workUnitForTable(HiveDataset hiveDataset) throws IOException {
```

Comment:
Can we just keep the older APIs and get rid of the methods where we pass disableAvroCheck as an argument? Instead, we can just access this property via the source state object.

Reply:
I did not do this because state is not a public variable. And this change dose not do this is because some of the downstream user has already change their code to enable the new API. And I also want to make this backward compatible with the latest version

Reply:
Do you mean to say that state is not a member variable? If so, we can create one. But I agree that reverting the newer APIs will be backward incompatible. Approving the RB.